### PR TITLE
Update CMUdict with ADLR version pronunciations

### PR DESCRIPTION
**Pull request by redoctopus**
Added and modified pronunciations can be found at the end of the file under the headings "Words added by NVIDIA" and "Pronunciation changes from 0.7b" respectively.

Signed-off-by: Jocelyn Huang <jocelynh@nvidia.com>

### CODE REVIEW

**Comment by Oktai15** [comment]
@redoctopus @blisc can we mark somehow not only added new words, but also edits (like monday) and removes (like 3D)?

**Comment by blisc** [comment]
> @redoctopus @blisc can we mark somehow not only added new words, but also edits (like monday) and removes (like 3D)?

Edits are marked in the last section.
Adding a comment section that lists all removes is a good idea.
The edit that reorder the different pronunciation choices (eg `monday`) would probably have to be done in-line. I can't think of a good way to separate them out.

**Comment by Oktai15** [comment]
@redoctopus @blisc okay, I got it. So, I like this PR, but let's move edits which are reordering the different pronunciation (DIGITAL, ENTREPRENEUR, FRIDAY, HUANG, MONDAY, REROUTING, SATURDAY, TUESDAY, WEDNESDAY) to `Pronunciation changes from 0.7b` block?

File: scripts/tts_dataset_files/cmudict-0.7b_nv22.01 [hunk 1]
```diff
@@ -133908,3 +133872,241 @@ ZYWICKI  Z IH0 W IH1 K IY0
...
+OMNIVERSE  AA1 M N IY0 V ER1 S
+OPENSTACK  OW1 P AH0 N S T AH1 K
+OPTIMIZED  AA1 P T IH0 M AY2 Z D
+OPTIMIZING  AA1 P T IH0 M AY2 Z IH0 NG
+PARABRICK  P EH1 R AH0 B R IH0 K
+PCB  P IY1 S IY1 B IY1
+PCI  P IY1 S IY1 EY1
+PCIE  P IY1 S IY1 EY1 IY1
+PERFORMANT  P ER0 F AO1 R M AH0 N T
+PETAFLOP P EH1 T AH0 F L AA2 P
```
Comment (blisc) [inline scripts/tts_dataset_files/cmudict-0.7b_nv22.01:134000]: This line is formatted incorrectly. Should be a double space after PETAFLOP not a single space

**Comment by blisc** [comment]
Let's remove this line `D�J�  D EY2 JH AA1`. It's crashing my train scripts

File: scripts/tts_dataset_files/cmudict-0.7b_nv22.01 [hunk 2]
```diff
@@ -133908,3 +133872,241 @@ ZYWICKI  Z IH0 W IH1 K IY0
...
+VIRTUALLIZED  V ER1 CH UW0 AH0 L AY2 Z D
+VMWARE  V IY2 EH2 M W EH1 R
+VXLAN  V IY1 EH1 K S L AE1 N
+WAVEGLOW  W EY1 V G L OW2
+WHEELHOUSE  W IY1 L HH AW2 S
+WJH  D AH1 B AH0 L Y UW0 JH EY1 EY1 CH
+;;;
+;;; Pronunciation changes from 0.7b
+;;;
+3D TH R IY1 D IY2
```
Comment (blisc) [inline scripts/tts_dataset_files/cmudict-0.7b_nv22.01:134070]: space issue here as well

File: scripts/tts_dataset_files/cmudict-0.7b_nv22.01 [hunk 3]
```diff
@@ -133908,3 +133872,241 @@ ZYWICKI  Z IH0 W IH1 K IY0
...
+A  EY1
+A(1)  AH1
+AC  EY2 S IY1
+AI  EY1 AY1
+AI(1) EY2 AY1
```
Comment (blisc) [inline scripts/tts_dataset_files/cmudict-0.7b_nv22.01:134075]: space issue here as well

File: scripts/tts_dataset_files/cmudict-0.7b_nv22.01 (hunk 1 above)
Comment (redoctopus) [reply scripts/tts_dataset_files/cmudict-0.7b_nv22.01:134000]: Thanks, good catch (x3).

**Comment by redoctopus** [comment]
> @redoctopus @blisc okay, I got it. So, I like this PR, but let's move edits which are reordering the different pronunciation (DIGITAL, ENTREPRENEUR, FRIDAY, HUANG, MONDAY, REROUTING, SATURDAY, TUESDAY, WEDNESDAY) to `Pronunciation changes from 0.7b` block too?

Sounds like a good idea to me, done.

**Review by Oktai15 — CHANGES_REQUESTED** [review]
Please, try to load CMUDict via https://github.com/NVIDIA/NeMo/blob/af18023b30e71262b27800ef75306de649cd6794/nemo/collections/tts/torch/g2ps.py#L96 and check that size of vocabulary equals to expected size

**Comment by redoctopus** [comment]
> Please, try to load CMUDict via
> 
> https://github.com/NVIDIA/NeMo/blob/af18023b30e71262b27800ef75306de649cd6794/nemo/collections/tts/torch/g2ps.py#L96
> 
> and check that size of vocabulary equals to expected size

The following works:
`d = g2ps.EnglishG2p._parse_as_cmu_dict(phoneme_dict_path='<NeMo path>/scripts/tts_dataset_files/cmudict-0.7b_nv22.01')`

Edited and added words are included in the result, except for the ones that start with numbers (e.g. 3D, 4K). There was a bit of a size mismatch from the total size of unique entries, but I think this is accounted for by the filtering/regex-squashing in the code.

**Review by Oktai15 — APPROVED** [review]

**Pull request merged**